lib/common/buffer-utils.ts

```diff
@@ -38,19 +33,25 @@ function getBuffer (value) {
   }
 }
 
-exports.isBuffer = function (value) {
+export function isBuffer (value: ArrayBuffer | ArrayBufferView) {
   return ArrayBuffer.isView(value) || value instanceof ArrayBuffer
 }
 
-exports.bufferToMeta = function (value) {
+interface BufferMeta {
+  type: keyof typeof typedArrays;
+  data: Buffer;
+  length: number;
+}
+
+export function bufferToMeta (value: ArrayBuffer | ArrayBufferView): BufferMeta {
   return {
     type: getType(value),
     data: getBuffer(value),
-    length: value.length
+    length: (value as Buffer).length
```

Comment:
this doesn't seem right. `ArrayBuffer` doesn't have a `length` property (it has `byteLength`).

Reply:
we only need `length` when we are serializing `Int8Array`, etc.
when deserializing `Buffer` and `ArrayBuffer`, `length` is not used

Reply:
Then `BufferMeta` should have `length: number | undefined`?

Reply:
Please also add a comment here, this is confusing & looks like a bug without the appropriate context.

Reply:
nvm i went ahead and added it :)